gui/webdisplay/src/TWebWindowsManager.cxx

```diff
@@ -707,7 +759,7 @@ bool ROOT::Experimental::TWebWindowsManager::Show(ROOT::Experimental::TWebWindow
 
    exec.ReplaceAll("$prog", prog.Data());
 
-   win.AddKey(key, where); // for now just application name
+   unsigned connid = win.AddProcId(batch_mode, key, where + rmdir); // for now just application name
```

Comment:
I understand that this is a convenient and http-style mechanism (strings everywhere). I'd nonetheless very much prefer to provide a "destruction callback" mechanism instead, where you pass in a lambda that unlinks the profile directory. Like that, all that profile handling remains in this function, instead of file system operations leaking into `HaltClient()`

Reply:
Please, in next PR. This object has to `travel` over many instances.